enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,179 @@
+
+## Motivation
+
+When managing the [Cincinnati][cincinnati-spec] update graph [for OpenShift][cincinnati-for-openshift-design], we sometimes discover stability issues with particular release images or updates between them.
+Once an issue is discovered, we [block edges][block-edges] so we no longer recommend risky updates, or updates to risky releases.
+
+Note: as described [in the documentation][support-documentation], supported updates are still supported even if incoming edges are blocked, and Red Hat will eventually provide supported update paths from any supported release to the latest supported release in its z-stream.
+
+Incoming bugs are evaluated to determine an impact statement based on [a generic template][rhbz-1858026-impact-statement-request].
+Some bugs only impact specific platforms, or clusters with other specific features.
+For example, rhbz#1858026 [only impacted][rhbz-1858026-impact-statement] clusters with the `None` platform which were created as 4.1 clusters and subsequently updated via 4.2, 4.3, and 4.4 to reach 4.5.
+In those cases there is currently tension between wanting to protect vulnerable clusters by blocking the edge vs. wanting to avoid inconveniencing clusters which we know are not vulnerable and whose administrators may have been planning on taking the recommended update.
+
+### Goals
+
+* [Cincinnati graph-data][graph-data] maintainers will have the ability to block edges for the subset of clusters matching a particular Prometheus query.
+
+### Non-Goals
+
+* Blocking edges based on data that is not [uploaded to Telemetry][uploaded-telemetry].
+* Blocking edges for subsets of clusters when the update service is unable to reach or authenticate with an aggregating Prometheus server.
+    For example, users running a local update service will not be able to access Red Hat's internal Telemetry to determine the requesting cluster's platform, etc.
+* Specifying a particular update service implementation.
+    This enhancement floats some ideas, but the details of the chosen approach are up to each update service's maintainers.
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The blocked-edges schema][block-edges] will be extended with the following new properties:
+
+* `clusters` (optional, [object][json-object]), defining the subset of affected clusters.
+    If any `clusters` property matches a given cluster, the edge should be blocked for that cluster.
+    * `prometheus` (optional, [object][json-string]), with a [PromQL][] query describing affected clusters.
+        This query will be submitted to a configurable Prometheus service and should return a set of matching records with `_id` labels.
+        Clusters whose [submitted `id` query parameter][cincinnati-for-openshift-request] is in the set of returned IDs are blocked if the value of the matching record is 1 and blocked if the value of the matching record is 0.
```

Comment:
```suggestion
        Clusters whose [submitted `id` query parameter][cincinnati-for-openshift-request] is in the set of returned IDs are blocked if the value of the matching record is 1 and allowed if the value of the matching record is 0.
```

Reply:
I was actually allowing on 1 and blocking on 0, in my later OSD example.  So fixed, but with the opposite sense to your patch, with 21d882c -> 228fa57.  Not sure if there's a more obvious way to declare the sense, I feel like it's a bit of a foot gun however you dice it.  Maybe ignore the value, and recommend folks use `label_replace` to set a `block` label to `true` or `false`?  I dunno, `label_replace` is such a pain...

Reply:
That looks good, thanks. Not sure if there is more intuitive way to describe it.
Perhaps we should slim it down to a promql query and introduce `type` with `allow_if_1` and `block_if_0` values or smth similar?

Reply:
I've pushed 1f64222 -> 2037152 with some more discussion of this choice in the new *Query coverage* section.